shub/deploy.py

```diff
@@ -187,6 +190,44 @@ def _get_pipfile_requirements():
     return convert_deps_to_pip(deps)
 
 
+def _is_poetry(name):
+    if name != 'pyproject.toml':
+        return False
+    data = toml.load(name)
+    return 'poetry' in (data.get('tool') or {})
+
+
+def _get_poetry_requirements():
+    try:
+        data = toml.load('poetry.lock')
+    except IOError:
+        raise ShubException('Please make sure the poetry lock file is present')
+    # Adapted from poetry 1.0.0a2 poetry/utils/exporter.py
+    with tempfile.NamedTemporaryFile(
+        dir=os.getcwd(), suffix='.txt', delete=False,
```

Comment:
I guess we have to remove the temporary file later somewhere, right? Or maybe return requirements content here instead of a filename and rework `_upload_egg` and skip `open()` call for it (it seems `requests` [allows](https://2.python-requests.org//en/latest/user/quickstart/#post-a-multipart-encoded-file) to pass strings instead of file objects).

Reply:
Yeah - actually I wasn't sure what was happening in the pipenv code in this case.  Is it creating a requirements.txt file?  Passing the string text itself seems like a nicer way to do it if you don't mind the change scope increasing.

Reply:
>  Is it creating a requirements.txt file? Passing the string text itself seems like a nicer way to do it if you don't mind the change scope increasing.

Yup, it is, and then it returns the temporary requirements filename to the caller which opens it and passes to requests call. It'd be a bit tricky to reliably drop the temporary file later, so passing a string makes sense to me. This is a helper, not a public API, also requirements files are not that large and we already load it to memory while constructing, so I believe it should be safe and cleaner to pass it via string object.